### ISSUE TITLE
the wrong loss func was chosen at evaluation

### ISSUE THREAD

**Issue by peterjc123** [stanford-futuredata/megablocks#93]
The loss func is always `moe_loss_func` as can be seen [here](https://github.com/stanford-futuredata/Megatron-LM/blob/3a9e3d8de308e6f6398b59d16a8bd7177374f121/pretrain_gpt.py#L128). But the loss is only calculated when training, which can be seen [here](https://github.com/stanford-futuredata/megablocks/blob/f05609ce69c1e1a7dd008c49cf435ef74df84b69/megablocks/layers/moe.py#L427-L428). We should fallback to the original loss func during evaluation.

**Comment by peterjc123** [comment]
@tgale96 I've put up a fix. Please take a look when you have time. https://github.com/stanford-futuredata/Megatron-LM/pull/6
